cmd/sonobuoy/app/gen.go

```diff
@@ -298,9 +298,22 @@ func mergeConfigs(dst, src *config.Config) {
 		emptyResources = true
 	}
 
+	// Workaround to differentiate a false value and a nil value
+	// Only override dst.Limits.PodLogs.SonobuoyNamespace when it's nil
+	// See https://github.com/imdario/mergo/issues/89
+	var sonobuoyNamespace *bool
+	if dst.Limits.PodLogs.SonobuoyNamespace != nil {
+		sonobuoyNamespace = new(bool)
+		*sonobuoyNamespace = *dst.Limits.PodLogs.SonobuoyNamespace
```

Comment:
TL;DR: 👍 

Long message: 

Frustrating to see something like this have to get an awkward implementation like this but I do understand the problem. For all the great stuff `mergo.Merge` can do I guess simple things like this make it clear that even though the idea is conceptually simple, the devil is in the details.

It may be fine as-is, but it could be resolved if we changed the field to something that inverted the default value. E.g. if we had `SkipSonobuoyNamespace` which defaulted to false, that would remove the need to do this workaround since we'd treat a nil object the same as the default, right? But that is more awkward to a user which is a higher priority than the simplicity of the implementation on something like this. :-/

Reply:
I have discussed this with @zubron, our concern is that setting it in inverted way will break the logic:
`(Namespaces OR SonobuoyNamespace OR FieldSelectors`

If user set:
- Namespaces = ".*"
- SkipSonobuoyNamespace = true

The thing becomes:
`(Namespaces OR FieldSelectors) AND SkipSonobuoyNamespace`, which :
- adds one more layer of different operator, and maybe confusing for the user.
- need to have specific code to check `SkipSonobuoyNamespace`, cannot just expand it to append to the fieldSelectors list. So we lost the generic way to handle the quick buttons as additional fieldSelectors, and the logic becomes a lot of if...else...

Reply:
Good point on the example 
 - Namespaces = ".*"
 - SkipSonobuoyNamespace = true

That is very confusing looking and seems like you're explicitly going to _not_ get the logs from the Sonobuoy namespace.

sgtm